### ISSUE TITLE
Required parameter error messages

### ISSUE THREAD

**Issue by cdibbs** [vvakame/commandpost#6]
Is it possible to setup a callback for missing required parameters? Right now, it shows a stack trace, by default. I would have to catch and parse the error message if I wanted something custom, right?

If this doesn't exist, my hope would be for something like this:
```typescript
            let newTemp = root
                .subCommand<INewCmdOpts, INewCmdArgs>("new <item>")
                .description("Generate a thing from an item.")
                .option("-n, --name <name>", "a description")
                .optionError((oName, errType, defaultMsg): string => customMessageIfAny)
                .commandError((cName, errType, defaultMsg): string => customMessageIfAny);
```

Thanks in advance.

**Comment by vvakame** [comment]
do you use [this code snippet?](https://github.com/vvakame/commandpost/blob/e1b96ff1367abe31e9bdd16061865e71fc944510/example/usage.ts#L71-L80)
You can use `console.error(err.message);` instead of `console.error(err.stack);`.
It does not output stack trace.

**Comment by cdibbs** [comment]
Ah, sorry. I was mistaken about the stack trace. Nevertheless, it would be nice to have a custom handling option. Maybe all that's needed is a subclass of Error which accepts two extra arguments:

```typescript
throw new ParseError(
    args[0] + " is required parameter value",
    args[0],
    ParseErrorType.OptionRequired);
```

Then, a user's catch block could look like this:
```typescript
    .catch(err => {
        if (err instanceof ParseError) {
            console.error(i18n.__mf(myCmdPostMsgs[err.type], err));
        } else if (err instanceof Error) {
            console.error(err.stack);
        } else {
            console.error(err);
        }
        process.exit(1);
    });
```

What do you think? If you're low on time, I could try to do this and submit a pull request.

**Comment by vvakame** [comment]
umm...

I can't make a good API design.
I tried custom error object to solve this issue.
Do you like this change?
https://github.com/vvakame/commandpost/commit/fc0dc871841a11389f63e62b1e524db3708de8ec

**Comment by vvakame** [comment]
@cdibbs I just published v1.3.0. please try it!

**Comment by cdibbs** [comment]
Works great! Thanks for your help, @vvakame.
